google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/CookieHolderTest.java

```diff
@@ -0,0 +1,157 @@
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.data.v2.stub;
+
+import static com.google.cloud.bigtable.data.v2.stub.CookiesHolder.ROUTING_COOKIE_METADATA_KEY;
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.gax.grpc.InstantiatingGrpcChannelProvider;
+import com.google.bigtable.v2.BigtableGrpc;
+import com.google.bigtable.v2.ReadRowsRequest;
+import com.google.bigtable.v2.ReadRowsResponse;
+import com.google.bigtable.v2.SampleRowKeysRequest;
+import com.google.bigtable.v2.SampleRowKeysResponse;
+import com.google.cloud.bigtable.data.v2.BigtableDataClient;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import com.google.cloud.bigtable.data.v2.FakeServiceBuilder;
+import com.google.cloud.bigtable.data.v2.models.Query;
+import com.google.common.collect.ImmutableList;
+import io.grpc.Metadata;
+import io.grpc.Server;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.StreamObserver;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class CookieHolderTest {
+  private Server server;
+  private FakeService fakeService = new FakeService();
+  private BigtableDataClient client;
+  private List<Metadata> serverMetadata = new ArrayList<>();
+  private String testCookie = "test-routing-cookie";
+
+  @Before
+  public void setup() throws Exception {
+    ServerInterceptor serverInterceptor =
+        new ServerInterceptor() {
+          @Override
+          public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
+              ServerCall<ReqT, RespT> serverCall,
+              Metadata metadata,
+              ServerCallHandler<ReqT, RespT> serverCallHandler) {
+            serverMetadata.add(metadata);
+            return serverCallHandler.startCall(serverCall, metadata);
+          }
+        };
+
+    server = FakeServiceBuilder.create(fakeService).intercept(serverInterceptor).start();
+
+    BigtableDataSettings.Builder settings =
+        BigtableDataSettings.newBuilderForEmulator(server.getPort())
+            .setProjectId("fake-project")
+            .setInstanceId("fake-instance");
+
+    InstantiatingGrpcChannelProvider channelProvider =
+        (InstantiatingGrpcChannelProvider) settings.stubSettings().getTransportChannelProvider();
+    settings
+        .stubSettings()
+        .setTransportChannelProvider(
+            channelProvider
+                .toBuilder()
+                .setInterceptorProvider(() -> ImmutableList.of(new CookieInterceptor()))
+                .build());
+
+    client = BigtableDataClient.create(settings.build());
+  }
+
+  @Test
```

Comment:
I think we need to future proof this so that when someone adds a method and forgets to add the cookie callable it will fail.

We should also ensure that responses arent dropped. So I'm thinking that your fake server needs an interceptor that records all of the MehtodDescriptors that were used and you have a single test that sends all of the rpcs. THen at the end of the test you diff that all of the MethodDescriptors are present.

Reply:
This doesnt look like it was addressed?